File: tests/journal.lib/file_example.cc
// -*- c++ -*-
//
// michael a.g. aïvázis <michael.aivazis@para-sim.com>
// (c) 1998-2023 all rights reserved


// get the journal
#include <pyre/journal.h>
// support
#include <cassert>


// alias the type
using debug_t = pyre::journal::debug_t;
using file_t = pyre::journal::file_t;


// exercise the file device
int main() {
    // the filename
    auto filename = file_t::path_type("file_example.out");

    // make a channel
    debug_t channel("test.journal.file");
    // set its device
    channel.device<file_t>(filename);
    // activate
    channel.activate();

    // inject something
    channel
        << pyre::journal::at(__HERE__)
        << "hello world!"
        << pyre::journal::endl;

    // all done
    return 0;
}


// end of file

File: tests/journal.pkg/file_example.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-
#
# michael a.g. aïvázis <michael.aivazis@para-sim.com>
# (c) 1998-2023 all rights reserved


def test():
    """
    Verify that we can send channel output to a log file
    """
    # pick a file name
    filename = "file_example.log"

    # get the device
    from journal.File import File
    # and a channel
    from journal.Debug import Debug

    # instantiate
    device = File(path=filename)

    # make a debug channel
    channel = Debug(name="tests.journal.debug")
    # activate it
    channel.activate()
    # send the output to the file
    channel.device = device

    # add some metadata
    channel.notes["time"] = "now"
    # inject
    channel.line("debug channel:")
    channel.log("    hello world!")

    # all done
    return


# main
if __name__ == "__main__":
    # prohibit the journal bindings
    journal_no_libjournal = True
    # run the test
    test()


# end of file
